pandas/core/construction.py

```diff
@@ -456,11 +456,29 @@ def sanitize_array(
     index: Optional[Index],
     dtype: Optional[DtypeObj] = None,
     copy: bool = False,
-    raise_cast_failure: bool = False,
+    raise_cast_failure: bool = True,
```

Comment:
assume you are trying to kill this parameter entirely

Reply:
doing that would require changing DataFrame constructor behavior, so for now im just trying to catch less